lib/elasticsearch.js

```diff
@@ -0,0 +1,540 @@
+			throw new ElasticSearchError(err.message, ElasticSearchError.codes.ELASTICSEARCH_ERROR);
+		}
+	}
+
+	/**
+	 * Get the paginated totals from latest get query.
+	 * @param {Model} model Model instance
+	 * @returns {Object} total, page size, pages and page from the results.
+	 */
+	async getTotals(model) {
+
+		this.validateModel(model);
+
+		if(model.lastQueryEmpty)
+			return { total: 0, pages: 0 };
+
+		const params = model.totalsParams || { filters: {} };
+
+		const filters = params.filters ? this.parseFilters(params.filters) : {};
+
+		try {
+			const client = this.getClient();
+			const res = await client.count({
+				index: model.constructor.table,
+				type: '_doc',
+				body: filters
+			});
+
+			const result = {
+				total: res.body.count,
+				pageSize: params.limit || this.config.limit,
+				pages: params.limit ? Math.ceil(res.body.count / params.limit) : 1,
+				page: params.page || 1
+			};
+
+			if(result.page > result.pages)
+				result.page = result.pages;
+
+			return result;
+
+		} catch(err) {
+			throw new ElasticSearchError(err.message, ElasticSearchError.codes.ELASTICSEARCH_ERROR);
+		}
+	}
+
+	/**
+	 * Updates data into the database
+	 * @param {Model} model Model instance
+	 * @param {Object} values values to apply
+	 * @param {Object} filter filters
+	 * @returns {Number} modified count
+	 */
+	async update(model, values, filters) {
+
+		this.validateModel(model);
+
+		try {
+			const client = this.getClient();
+			const res = await client.updateByQuery({
+				index: model.constructor.table,
+				refresh: true,
+				type: '_doc',
+				body: {
+					...this.parseFilters(filters),
+					...this.parseValuesForUpdate(values)
+				}
+			});
+
+			return res.body.updated;
+
+		} catch(err) {
+			throw new ElasticSearchError(err.message, ElasticSearchError.codes.ELASTICSEARCH_ERROR);
+		}
+	}
+
+	/**
+	 * Insert/update one element into the database
+	 * @param {Model} model Model instance
+	 * @param {Object} item item
+	 * @returns {Boolean} true/false
+	 */
+	async save(model, item) {
+
+		this.validateModel(model);
+		this.prepareFields(item);
+
+		try {
+			const client = this.getClient();
+			const res = await client.update({
+				index: model.constructor.table,
+				type: '_doc',
+				id: item.id,
+				refresh: 'wait_for',
+				body: {
+					doc: { ...item, lastModified: new Date() },
+					upsert: { ...item, dateCreated: new Date() }
+				}
+
+			});
+
+			return res.body.result === 'created' || res.body.result === 'updated';
+
+		} catch(err) {
+			throw new ElasticSearchError(err.message, ElasticSearchError.codes.ELASTICSEARCH_ERROR);
+		}
+	}
+
+	/**
+	 * Multi insert/update items into the dabatase
+	 * @param {Model} model Model instance
+	 * @param {Array} items items
+	 * @param {Number} limit specifies the limit of items that can be bulk writed into monogdb at the same time
+	 * @returns {Boolean} true/false
+	 */
+	async multiSave(model, items) {
+
+		this.validateModel(model);
+		this.prepareFields(items);
```

Comment:
Se podria hacer que retorne el valor modificado en vez de modificarlo por dentro
```suggestion
		item = this.prepareFields(item);
```